### ISSUE TITLE
Add support for windows

### ISSUE THREAD

**Issue by miversen33** [miversen33/netman.nvim#82]
Not really sure how I will go about doing this, and the current interest in it is basically none. But at some point I should probably consider doing this.

**Comment by miversen33** [comment]
Fun fact, even attempting to load netman on windows causes it to just lock up all of neovim. So thats great :upside_down_face:

**Comment by miversen33** [comment]
I have added basic core support for Windows. It seems there is going to be some shenanigans for supporting windows within the providers so I will need to figure out how to abstract that garbage to make it so providers don't need to care as much...

Current errors I am getting are below (this will be edited several times over)

# SSH
- [ ] getsockname failed: Bad file descriptorchannel_send_open: channel 0: unexpected internal error

# Docker

**Comment by miversen33** [comment]
A potential way to handle some of the awful performance issues with ssh on windows is to (potentially) pin a `/bin/sh` shell to the `ssh` object and simply execute all commands over that. This appears to be reasonably possible. Below is a snippet of code showcasing that the inbuilt `shell` object can handle this (though we still have a slight issue with password auth???)

```lua
package.loaded['netman.tools.shell'] = nil
local SHELL = require("netman.tools.shell")

local shell = nil
local handle = nil

local stdout = function(data)
    io.write(data)
end

local stderr = function(data)
    io.write(string.format("ERROR: ", data))
end

local function get_input()
    if not handle.is_active then
        return false
    end
    io.write("$ ")
    local input = io.read()
    if input:match('^exit$') then
        handle.stop()
        return false
    end
    if not input:match('$[\r\n]') then
        input = input .. '\n'
    end
    handle.write(input)
    return true
end

local command = {"/usr/bin/ssh", "-t", "dev-miversen", "/bin/zsh", "-l"}
shell = SHELL:new(command, {
    [SHELL.CONSTANTS.FLAGS.ASYNC] = true,
    [SHELL.CONSTANTS.FLAGS.STDOUT_CALLBACK] = stdout,
    [SHELL.CONSTANTS.FLAGS.STDERR_CALLBACK] = stderr
})
handle = shell:run(-1)

-- Loop until 🤷
while get_input() do
    require("luv").run('once')
end
```

This guy will successfully connect to the `dev-miversen` server, open a zsh shell and pin it allowing me (the user) to talk to it, execute commands over it, etc. This is super promising as it potentially solves the Windows perf issue, #33, and #66 all at once :)

**Comment by miversen33** [comment]
This has been marinating for a while and while I haven't solved the windows perf issues, I do have basic windows compatibility solved. I am going to merge this into main (and pull it into the various dev branches). I will look to do that sometime this weekend and there _will_ be a new release to accompany this.

This will _not_ be the `1.15.0` release, rather `1.10.01`

**Comment by miversen33** [comment]
While support for windows isn't _great_, it does work. Merged into v1.15 and pulled into the Async branch as well.
